### PR TITLE
Refactor FXIOS-9034 Remove redundant private tab code

### PR DESCRIPTION
**Pull request by OrlaM**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9034)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19952)

## :bulb: Description
Cleans up code related to private tab storage

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Coordinators/Router/RouteBuilder.swift [hunk 1]
```diff
@@ -202,7 +196,7 @@ final class RouteBuilder {
             TelemetryWrapper.recordEvent(
                 category: .action,
                 method: .open,
-                object: isPrivate ?.mediumQuickActionPrivateSearch:.mediumQuickActionSearch
```
Comment (cyndichin) [inline firefox-ios/Client/Coordinators/Router/RouteBuilder.swift:205]: should we also remove any mention of `mediumQuickActionPrivateSearch` since it's not longer being used?

File: firefox-ios/Client/Coordinators/Router/RouteBuilder.swift [hunk 2]
```diff
@@ -63,7 +57,7 @@ final class RouteBuilder {
             case .openUrl:
                 // If we have a URL query, then make sure to check its a webpage
                 if urlQuery == nil || urlQuery?.isWebPage() ?? false {
-                    return .search(url: urlQuery, isPrivate: isPrivate)
+                    return .search(url: urlQuery, isPrivate: false)
```
Comment (cyndichin) [inline firefox-ios/Client/Coordinators/Router/RouteBuilder.swift:60]: for my better understanding, can you help clarify why this change and why only these places are set to false?

**Review by cyndichin — APPROVED** [review]
looks good, one comment for me to better understand and a small suggestion for more cleanup. thanks!

Seems build is failing due to tests not being updated to account for the changes.

File: firefox-ios/Client/Coordinators/Router/RouteBuilder.swift (hunk 2 above)
Comment (OrlaM) [reply firefox-ios/Client/Coordinators/Router/RouteBuilder.swift:60]: Previously they would open the app into private mode if the previous session was a private session. But since we no longer store the private tab data the expected session type on launch is normal browsing (unless the user explicitly uses the widget or shortcut to open a private tab).

Comment (mattreaganmozilla) [reply firefox-ios/Client/Coordinators/Router/RouteBuilder.swift:60]: > since we no longer store the private tab data the expected session type on launch is normal browsing

Some of these codepaths are also called when the app is simply in the background, however (as opposed to a cold launch). And in those cases private tabs are still active. The widget routing can run with private tabs open and at least in a quick test I could see this .`openURL` case handled when a private tab was still open/active. At first glance I don't think we should be hardcoding these to `false` 🤔

Comment (OrlaM) [reply firefox-ios/Client/Coordinators/Router/RouteBuilder.swift]: Yeah I misunderstood how this works, the tests caught the issue. I've just pushed an update with this fixed. It no longer passes in the previous private session but does still look through the url for the private parameter.

**Comment by OrlaM** [comment]
@cyndichin This needs another review pass. I've fixed the gap with the route builder and I removed some more code I missed before.

File: firefox-ios/Client/Coordinators/Router/RouteBuilder.swift [hunk 3]
```diff
@@ -126,7 +123,7 @@ final class RouteBuilder {
             TelemetryWrapper.gleanRecordEvent(category: .action, method: .open, object: .asDefaultBrowser)
             RatingPromptManager.isBrowserDefault = true
             // Use the last browsing mode the user was in
-            return .search(url: url, isPrivate: isPrivate, options: [.focusLocationField])
+            return .search(url: url, isPrivate: false, options: [.focusLocationField])
```
Comment (cyndichin) [inline firefox-ios/Client/Coordinators/Router/RouteBuilder.swift:126]: do we still want this as `false`?

File: firefox-ios/Client/Application/SceneDelegate.swift [hunk 4]
```diff
@@ -42,8 +42,7 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
             Experiments.shared.initializeTooling(url: url)
         }
 
-        routeBuilder.configure(isPrivate: UserDefaults.standard.bool(forKey: PrefsKeys.LastSessionWasPrivate),
```
Comment (cyndichin) [inline firefox-ios/Client/Application/SceneDelegate.swift:45]: do we want to remove `LastSessionWasPrivate` from the prefkeys as well?

Comment (OrlaM) [reply firefox-ios/Client/Application/SceneDelegate.swift:45]: It is removed in this PR.

File: firefox-ios/Client/Coordinators/Router/RouteBuilder.swift (hunk 3 above)
Comment (OrlaM) [reply firefox-ios/Client/Coordinators/Router/RouteBuilder.swift:126]: I believe so, this case previously ignored any parameters in the url and relied solely on the pref for if the last session was private.

**Review by cyndichin — APPROVED** [review]
looks good! thanks @OrlaM !
